### PR TITLE
[WIP] Refactor api and frontend logic for firebase auth

### PR DESCRIPTION
**Pull request by williamchong007**

### CODE REVIEW

File: pages/in/settings/others.vue [hunk 1]
```diff
@@ -136,6 +132,7 @@ export default {
       this.isLoading = true;
       try {
         const user = this.getUserInfo;
+        console.log(typeof this.isEmailEnabled);
```
Comment (williamchong007) [inline pages/in/settings/others.vue]: 👎

File: store/modules/actions/api-wrapper.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ async function apiWrapper({ commit, dispatch }, promise, opt = {}) {
     const { response } = error;
     if (response && response.statusCode === 401) {
       await dispatch('doUserAuth');
+      console.log('doUserAuth');
```
Comment (williamchong007) [inline store/modules/actions/api-wrapper.js]: 👎

File: server/api/users.js [hunk 3]
```diff
@@ -351,6 +360,7 @@ router.get('/users/self', jwtAuth('read'), async (req, res, next) => {
       res.sendStatus(404);
     }
   } catch (err) {
+    console.error(err);
```
Comment (williamchong007) [inline server/api/users.js]: 👎

**Comment by nwingt** [comment]
Merged in #782

**Pull request closed without merging**